terraform/modules/spack/monitoring_db.tf

```diff
@@ -0,0 +1,440 @@
+
+  db_name                = "gitlabhq_production"
+  username               = "postgres"
+  port                   = "5432"
+  create_random_password = false
+  password               = local.gitlab_db_clone_master_password
+
+  publicly_accessible  = false
+  db_subnet_group_name = aws_db_subnet_group.spack.name
+
+
+  maintenance_window              = "Sun:00:00-Sun:03:00"
+  backup_window                   = "03:00-06:00"
+  enabled_cloudwatch_logs_exports = ["postgresql", "upgrade"]
+  create_cloudwatch_log_group     = true
+
+  backup_retention_period = 7
+  skip_final_snapshot     = false
+  deletion_protection     = true
+
+  allocated_storage     = 500
+  max_allocated_storage = 1000
+
+  vpc_security_group_ids = [module.postgres_security_group.security_group_id]
+}
+
+
+# Defines the Change Data Capture migration between the source and clone RDS instances
+module "database_migration_service" {
+  count = var.provision_monitoring_db ? 1 : 0
+
+  source  = "terraform-aws-modules/dms/aws"
+  version = "~> 1.6"
+
+  # Subnet group
+  repl_subnet_group_name        = "dms-subnet-group"
+  repl_subnet_group_description = "DMS Subnet group"
+  repl_subnet_group_subnet_ids  = module.vpc.private_subnets
+
+  # Instance
+  repl_instance_id                           = "gitlab-db-sync-replication-instance"
+  repl_instance_allocated_storage            = 64
+  repl_instance_auto_minor_version_upgrade   = true
+  repl_instance_allow_major_version_upgrade  = true
+  repl_instance_apply_immediately            = true
+  repl_instance_engine_version               = "3.4.7" # Must use 3.4.7 to support Postgres 14.x
+  repl_instance_multi_az                     = true
+  repl_instance_preferred_maintenance_window = "sun:10:30-sun:14:30"
+  repl_instance_publicly_accessible          = false
+  repl_instance_class                        = "dms.t3.large"
+
+  # Endpoints
+  endpoints = {
+    source = {
+      # Required
+      endpoint_id   = "gitlab-db-source-endpoint"
+      endpoint_type = "source"
+      engine_name   = "postgres"
+      # Optional
+      database_name = module.gitlab_db.db_instance_name
+      username      = module.gitlab_db.db_instance_username
+      password      = local.gitlab_db_master_password
+      port          = module.gitlab_db.db_instance_port
+      server_name   = module.gitlab_db.db_instance_address
+      tags          = { EndpointType = "source" }
+    }
+
+    destination = {
+      # Note: It's difficult to encode this into terraform at the moment (https://github.com/hashicorp/terraform-provider-aws/issues/23225),
+      # but the following setting needs to be entered into the "Endpoint Settings" of the target endpoint
+      # `AfterConnectScript="SET session_replication_role=replica"`
+
+      # This can also be done with the following command from the CLI:
+      # aws dms modify-endpoint --endpoint-arn 'ENDPOINT_ARN' --postgre-sql-settings='AfterConnectScript="SET session_replication_role=replica"'
```

Comment:
:disappointed: that's unfortunate, guess we should keep an eye on that issue

Reply:
Yeah, aside from some very convoluted and brittle configuration, I'm not sure what else we can do.